src/ControllerMessenger.ts

```diff
@@ -158,4 +247,17 @@ export class ControllerMessenger<Action extends ActionConstraint, Event extends
   clearSubscriptions() {
     this.events.clear();
   }
+
+  getRestricted<N extends string, AllowedAction extends string, AllowedEvent extends string>(
```

Comment:
All of these new methods need docs.

Also I'm going to change these params to take an options object instead, to improve readability.

Reply:
Done.